exporter/opentelemetry-exporter-prometheus-remote-write/README.rst

```diff
@@ -1,27 +1,248 @@
+   )
+
+   # Sets the global MeterProvider instance
+   metrics.set_meter_provider(MeterProvider())
+
+   # The Meter is responsible for creating and recording metrics. Each meter has a unique name, which we set as the module's name here.
+   meter = metrics.get_meter(__name__)
+
+   exporter = PrometheusRemoteWriteMetricsExporter(endpoint="endpoint_here") # add other params as needed
+
+   metrics.get_meter_provider().start_pipeline(meter, exporter, 5)
+
+Configuring the Exporter
+------------------------
+
+The exporter can be configured through parameters passed to the
+constructor. Here are all the options:
+
+-  ``endpoint``: url where data will be sent **(Required)**
+-  ``basic_auth``: username and password for authentication
+   **(Optional)**
+-  ``headers``: additional headers for remote write request as
+   determined by the remote write backend's API **(Optional)**
+-  ``timeout``: timeout for requests to the remote write endpoint in
+   seconds **(Optional)**
+-  ``proxies``: dict mapping request proxy protocols to proxy urls
+   **(Optional)**
+-  ``tls_config``: configuration for remote write TLS settings
+   **(Optional)**
+
+Example with all the configuration options:
+
+.. code:: python
+
+   exporter = PrometheusRemoteWriteMetricsExporter(
+       endpoint="http://localhost:9009/api/prom/push",
+       timeout=30,
+       basic_auth={
+           "username": "user",
+           "password": "pass123",
+       },
+       headers={
+           "X-Scope-Org-ID": "5",
+           "Authorization": "Bearer mytoken123",
+       },
+       proxies={
+           "http": "http://10.10.1.10:3000",
+           "https": "http://10.10.1.10:1080",
+       },
+       tls_config={
+           "cert_file": "path/to/file",
+           "key_file": "path/to/file",
+           "ca_file": "path_to_file",
+           "insecure_skip_verify": true, # for developing purposes
+       }
+   )
+
+Securing the Exporter
+---------------------
+
+Authentication
+~~~~~~~~~~~~~~
+
+The exporter provides two forms of authentication which are shown below.
+Users can add their own custom authentication by setting the appropriate
+values in the ``headers`` dictionary
+
+1. Basic Authentication Basic authentication sets a HTTP Authorization
+   header containing a base64 encoded username/password pair. See `RFC
+   7617`_ for more information. This
+
+.. code:: python
+
+   exporter = PrometheusRemoteWriteMetricsExporter(
+       basic_auth={"username": "base64user",  "password": "base64pass"}
+   )
+
+2. Bearer Token Authentication This custom configuration can be achieved
+   by passing in a custom ``header`` to the constructor. See `RFC 6750`_
+   for more information.
+
+.. code:: python
+
+   header = {
+       "Authorization": "Bearer mytoken123"
+   }
+
+TLS
+~~~
+
+Users can add TLS to the exporter's HTTP Client by providing certificate
+and key files in the ``tls_config`` parameter.
+
+Supported Aggregators
+---------------------
+
+-  Sum
```

Comment:
I think exposing the supported aggregators is great. Just wondering if we should actually just link to the metrics sdk definitions, so we don't have to always update this file when the sdk supports more aggregations.

Reply:
Due to differences between the aggregator data model  and Prometheus's `TimeSeries` definition, each aggregator is actually [converted separately](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/master/exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py#L75)  to the`TimeSeries` format. So even if the SDK adds more aggregators, this exporter might not be able to work with them unless a conversion method is added here :/ 

That's why I thought it would be worth explicitly listing out which aggregators are supported

Reply:
It would be nice to include a short description of what how these aggregators will behave.

Reply:
Thats a good idea, I added a link to the relevant section in the metric spec instead in case the behavior is subject to change

Reply:
I thought the point of listing out the aggregators was because they are the ones that Prometheus supports? We should be linking to Prometheus documentation of supported aggregators and what they do, instead of linking to OpenTelemetry documentation.

Reply:
We are exporting OTLP metrics to Prometheus remote write integrated back-ends in timeseries format so despite the name, we never actually touch Prometheus metrics any step along the way, we bypass the Prometheus format. The only link is that our timeseries are in the same format Prometheus metrics would be if they were exported to these back-ends. We thought users would be interested to know how exactly their OTLP aggregations would be once converted to timeseries since that is what they will see on the back-end. We could also link to the Prometheus documentation that guide us on the timeseries format, but I think the OTLP aggregation documentation is important to understand what the exporter did to a user's metrics.

Reply:
as per our meeting today, I've added a table comparing the OTLP aggegator with its equivalent Prometheus data type